engine/class_modules/sc_death_knight.cpp

```diff
@@ -6809,6 +6823,12 @@ struct vampiric_blood_buff_t : public buff_t
     // Cooldown handled by the action
     cooldown -> duration = 0_ms;
     base_buff_duration += player -> spec.vampiric_blood_2 -> effectN( 3 ).time_value();
+    if ( player -> dbc -> ptr )
+    {
+      set_default_value_from_effect( 5 );
+      apply_affecting_aura( player -> legendary.vampiric_aura );
+      add_invalidate( CACHE_LEECH );
+    }
```

Comment:
I wouldn't invalidate leech cache if the legendary isn't equipped/the default value is still 0. It also feels a bit weird to use the buff's default value to store what is just a side-effect from a legendary.

Reply:
The number 5 should also bei either retrieved from spell data, or otherwise add a comment.

I agree that misusing the buff value is a bit weird, but I guess it works. Until the normal buff wants to use that value :D

Reply:
The value is retrieved from spelldata, in effect #5, via the use of set_default_value_from_effect(). It starts at 0 and is modified by the legendary vampiric aura data.

Reply:
Right, I should have read that more carefully.